From 3ebf11b0ec323330e112dd535c6d99991777523e Mon Sep 17 00:00:00 2001
From: sai surya teja <sssteja@habana.ai>
Date: Tue, 07 Nov 2023 14:14:01 +0200
Subject: [PATCH] [SW-114485] gaudi3: add preboot die1 image pointer support

This patch adds preboot die1 image pointer support.
Preboot Die1 present at SPI offset Sector 20 (0x140000)
Die0 preboot FW does SPI DMA first to D0_ARC1_ICCM and later does
mem_copy to D1_ARC1_ICCM. This is needed because Die0 SPI DMA cannot
copy to Die1 memories.

Change-Id: I560f3a425e2c4c2bc4d21ea7e38b9220cea3737b
---

diff --git a/board/habanalabs/gaudi3_arc/post-image-spi.sh b/board/habanalabs/gaudi3_arc/post-image-spi.sh
index e4679b1..50b5e2c 100755
--- a/board/habanalabs/gaudi3_arc/post-image-spi.sh
+++ b/board/habanalabs/gaudi3_arc/post-image-spi.sh
@@ -30,12 +30,21 @@
 PREBOOT2_OFFSET=0xC0000
 PREBOOT_TEXT_BASE=0x0200FFFFFD000000
 
+# Preboot Die1 present at SPI offset 0x140000
+# Die0 preboot FW does SPI DMA first to D0_ARC1_ICCM and later does mem_copy to D1_ARC1_ICCM
+# This is needed because Die0 SPI DMA cannot copy to Die1 memories
+PREBOOT_D1_OFFSET=0x140000
+PREBOOT_D12_OFFSET=$PREBOOT_D1_OFFSET
+PREBOOT_D1_TEXT_BASE=0x0300007FD3040000
+PREBOOT_D1_START_ADDR=$PREBOOT_D1_TEXT_BASE
+
 EROM_PCIE_BINFILE=${EROM_PCIE_BINFILE:-erom_pcie.bin}
 EROM_RECOVERY_BINFILE=${EROM_RECOVERY_BINFILE:-erom_recovery.bin}
 
 EROM_PCIE_PADDED_BINFILE=${EROM_PCIE_PADDED_BINFILE:-${DEV_PREFIX}-erom_pcie.bin}
 EROM_RECOVERY_PADDED_BINFILE=${EROM_RECOVERY_PADDED_BINFILE:-${DEV_PREFIX}-erom_recovery.bin}
 PREBOOT_PADDED_BINFILE=${PREBOOT_PADDED_BINFILE:-preboot.bin.pad}
+PREBOOT_D1_PADDED_BINFILE=${PREBOOT_D1_PADDED_BINFILE:-preboot_d1.bin.pad}
 
 EROM_PCIE_OFFSET=0x1000
 EROM_RECOVERY_OFFSET=0x3000
@@ -370,7 +379,7 @@
 		img_name_2="PPBOOT_PCIE_PHY"
 	fi
 
-	if [ $img_name == "EROM_PCIE" -o $img_name == "EROM_RECOVERY" -o $img_name == "PREBOOT" ]
+	if [ $img_name == "EROM_PCIE" -o $img_name == "EROM_RECOVERY" -o $img_name == "PREBOOT" -o $img_name == "PREBOOT_D1" ]
 	then
 		img_name_2=${img_name}_PADDED
 	fi
@@ -459,7 +468,8 @@
 
 	local ipnames=0
 	local imgs_num_corr=0
-	local ipnames_spi_img=("PREBOOT" "UBOOT" "PPBOOT" "FAKE_FUSE" "PIF_ICCM" "PIF_DCCM")
+
+	local ipnames_spi_img=("PREBOOT" "UBOOT" "PPBOOT" "FAKE_FUSE" "PIF_ICCM" "PIF_DCCM" "PREBOOT_D1")
 	local ipnames_erom=("EROM_PCIE" "EROM_RECOVERY")
 	# Image pointers
 	if [ ${EROM_IMG} == 0 ]
